File: engine/transpile/src/test/resources/golden/SYK003/LK_チェック結果.py
"""SYK003 のデータ部レコード LK-チェック結果 を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class LK_チェック結果:
    """01 LK-チェック結果 総バイト長 1。"""

    LENGTH = 1

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(1)
        else:
            self._data = bytearray(data)
            if len(self._data) < 1:
                self._data.extend(b"\x00" * (1 - len(self._data)))

    @property
    def data(self):
        return self._data

    # LK-チェック結果
    def get_LK_チェック結果(self):
        return decode_alphanumeric(self._data, 0, 1)
    def set_LK_チェック結果(self, value):
        encode_alphanumeric(self._data, 0, 1, value)

File: src/engine/transpile/src/test/resources/golden/SYK003/LK_チェック結果.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK003 のデータ部レコード LK-チェック結果 を逐語対訳した自動生成コード。 */
public final class LK_チェック結果 {

    /** 01 LK-チェック結果 総バイト長 1。 */
    public static final int LENGTH = 1;

    private final byte[] data;

    public LK_チェック結果() {
        this.data = new byte[1];
    }

    public LK_チェック結果(byte[] source) {
        this.data = Arrays.copyOf(source, 1);
    }

    public byte[] data() {
        return data;
    }

    // LK-チェック結果
    public String get_LK_チェック結果() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 1);
    }
    public void set_LK_チェック結果(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 1, value);
    }
}
